client/asset/zec/zec.go

```diff
@@ -259,6 +256,11 @@ type zecWallet struct {
 	lastAddress atomic.Value // "string"
 }
 
+// FeeRate returns the asset standard fee rate for Zcash.
+func (w *zecWallet) FeeRate(context.Context) (uint64, error) {
```

Comment:
I think this is incorrect.  If the goal is to satisfy a `client/asset.FeeRater`, this does not do it.

I started writing a test like this:

```go
func TestFeeRate(t *testing.T) {
	wi, err := NewWallet(&asset.WalletConfig{
		Type: walletTypeRPC,
		Settings: map[string]string{
			"rpcuser":     "user",
			"rpcpassword": "pass",
			"rpcport":     "33770",
		},
		PeersChange: func(uint32, error) {},
		TipChange:   func(err error) {},
	}, dex.StdOutLogger("tst", dex.LevelDebug), dex.Simnet)
	if err != nil {
		t.Fatal(err)
	}

	ctx, cancel := context.WithCancel(context.Background())
	cancel() // context must be ignored
	rate, err := wi.(*zecWallet).FeeRate(ctx)
	if err != nil {
		t.Fatal(err)
	}
	if rate != 11 {
		t.Error("wrong")
	}
}
```

But then went to do `wi.(asset.FeeRater).FeeRate` and realized it didn't apply.

Reply:
Noticed something related:

```go
var _ asset.FeeRater = (*ExchangeWalletNoAuth)(nil) // FAIL, does not implement FeeRater
```

because

```go
func (btc *ExchangeWalletFullNode) FeeRate() uint64 {
```

I'm pretty certain that's a mistake.  It should be a `intermediaryWallet` method.

Reply:
Well, I really don't want to make another layer of intermediary wallet types for the sake of making `ExchangeWalletNoAuth` a `FeeRater`, so Iet's just dup the method and add the assertion:

```go
var _ asset.FeeRater = (*ExchangeWalletFullNode)(nil)
var _ asset.FeeRater = (*ExchangeWalletNoAuth)(nil)

func (btc *intermediaryWallet) feeRaterHelper() uint64 {
	// NOTE: With baseWallet having an optional external fee rate source, we may
	// consider making baseWallet a FeeRater by allowing a nil local func.
	rate, err := btc.feeRate(1)
	if err != nil {
		btc.log.Tracef("Failed to get fee rate: %v", err)
		return 0
	}
	return rate
}

// FeeRate satisfies asset.FeeRater.
func (btc *ExchangeWalletFullNode) FeeRate() uint64 {
	return btc.feeRaterHelper()
}

// FeeRate satisfies asset.FeeRater.
func (btc *ExchangeWalletNoAuth) FeeRate() uint64 {
	return btc.feeRaterHelper()
}
```

Reply:
Nothing wrong with it, but I'm just wondering why does `ExchangeWalletNoAuth` need to be a `FeeRater`? In the ZCash `NewWallet` function we return the `zecWallet` type.

Reply:
Just a related bug.